src/pres/loader/PGNLoader.js

```diff
@@ -16,36 +16,39 @@ export default class PGNLoader extends React.Component {
     constructor(props) {
         super(props)
         let selectedSite = new URLSearchParams(window.location.search).get("source")
+        let playerName = new URLSearchParams(window.location.search).get("playerName")
+        let color = new URLSearchParams(window.location.search).get("color") // 'white' | 'black'
+        // 'all' | 'rated' | 'casual' (corresponds to Filters.toggleRated)
+        let ratedMode = new URLSearchParams(window.location.search).get("ratedMode")
+        let fromDate = new URLSearchParams(window.location.search).get("fromDate") // YYYY-MM-DD
+        let toDate = new URLSearchParams(window.location.search).get("toDate") // YYYY-MM-DD
+        let timeControls = this.getTimeControls()
 
         this.state = {
-            playerName: '',
+            playerName: playerName?playerName:'',
             site: selectedSite?selectedSite:'',
-            playerColor: this.props.settings.playerColor,
+            playerColor: color ? color : this.props.settings.playerColor,
```

Comment:
Oh -- I don't know what `this.props.settings` is, actually! So might be good to double check if this fallback behavior makes sense.

Reply:
settings should ideally contain things related to the app itself (board orientation, dark mode etc). but today it has things like `playername` and `playercolor`. having two places for playercolor/playername is actually not good.

This might require a bunch of refactoring but we should rename your 'urlSettings' to something like `gameSettings` and then only use that going forward for storing info that users provide to load games.

Reply:
Makes sense! I'll poke around and ask you for some help and/or guidance here probably